src/v/serde/serde.h

```diff
@@ -729,7 +729,7 @@ void read_nested(iobuf_parser& in, T& t, std::size_t const bytes_left_limit) {
         in.consume_to(str.size(), str.begin());
         t = str;
     } else if constexpr (std::is_same_v<Type, uuid_t>) {
-        in.consume_to(uuid_t::length, t.uuid.begin());
+        in.consume_to(uuid_t::length, t.uuid().begin());
```

Comment:
could you remove the mutable accessor to uuid() and do something like

```
uuid_t::data data;
in.consume_to(uuid_t::length, data.begin());
t = uuid_t(std::move(data));
```

or some such thing?

Reply:
Yeah, though worth noting it incurs an extra copy since the underlying type is trivially copyable (uint8_t[16]).

Reply:
Chatted briefly on Slack/zoom -- I went the route of keeping the accessor but calling it `mutable_uuid()` so it's harder to accidentally misuse.